src/mesh/checkpoint_io.C

```diff
@@ -523,82 +510,129 @@ void CheckpointIO::read (const std::string & name)
 
   MeshBase & mesh = MeshInput<MeshBase>::mesh();
 
-  // Try to dynamic cast the mesh to see if it's a DistributedMesh object
-  // Note: Just using is_serial() is not good enough because the Mesh won't
-  // have been prepared yet when is when that flag gets set to false... sigh.
-  _parallel = _parallel || !mesh.is_replicated();
+  // Will this be a parallel input file?  With how many processors?  Stay tuned!
+  unsigned int input_parallel;
+  processor_id_type input_n_procs;
 
-  // If this is a serial mesh then we're going to only read it on processor 0 and broadcast it
-  if (_parallel || _my_processor_id == 0)
+  // We'll write a header file from processor 0 and broadcast.
+  if (this->processor_id() == 0)
     {
-      std::ostringstream file_name_stream;
-
-      file_name_stream << name;
-
-      if (_parallel)
-        file_name_stream << "-" << _my_n_processors << "-" << _my_processor_id;
-
       {
-        std::ifstream in (file_name_stream.str().c_str());
+        std::ifstream in (name);
```

Comment:
Hmm...  this is too big a change set to backport to the release branch, huh?  But when we do add C++11 code it ought to be because we need it, and we ought to add a configure-time warning about it.  I'll make this C++03-safe.